src/Scaffolding/VS.Web.CG.Mvc/Blazor/BlazorWebCRUDGenerator.cs

```diff
@@ -0,0 +1,352 @@
+        /// <param name="model"></param>
+        /// <returns></returns>
+        public async Task GenerateCode(BlazorWebCRUDGeneratorCommandLineModel model)
+        {
+            model.ValidateCommandline();
+            var emptyTemplate = !string.IsNullOrEmpty(model.TemplateName) && model.TemplateName.Equals("empty", StringComparison.OrdinalIgnoreCase);
+            if (emptyTemplate)
+            {
+                ArgumentNullException.ThrowIfNull(model);
+                string emptyRazorName = "Empty";
+                var outputPath = ValidateAndGetOutputPath(string.Empty, emptyRazorName);
+
+                //arguments for 'dotnet new razorcomponent'
+                var additionalArgs = new List<string>()
+                {
+                    "razorcomponent",
+                    "--name",
+                    emptyRazorName,
+                    "--output",
+                    Path.GetDirectoryName(outputPath),
+                };
+
+                DotnetCommands.ExecuteDotnetNew(ProjectContext.ProjectFullPath, additionalArgs, Logger);
+                return;
+            }
+
+            //get model and dbcontext
+            var modelTypeAndContextModel = await ModelMetadataUtilities.GetModelEFMetadataBlazorAsync(
+                model,
+                EntityFrameworkService,
+                ModelTypesLocator,
+                Logger,
+                areaName: string.Empty);
+
+            //check if getting model and dbcontext was successfull
+            if (modelTypeAndContextModel is null ||
+                modelTypeAndContextModel.ContextProcessingResult is null ||
+                modelTypeAndContextModel.ContextProcessingResult.ContextProcessingStatus is ContextProcessingStatus.MissingContext ||
+                modelTypeAndContextModel.ModelType is null)
+            {
+                throw new InvalidOperationException("Unable to get model and/or dbcontext metadata.");
+            }
+
+            if (!string.IsNullOrEmpty(modelTypeAndContextModel.DbContextFullName) && CalledFromCommandline)
+            {
+                EFValidationUtil.ValidateEFDependencies(ProjectContext.PackageDependencies, model.DatabaseProvider);
+            }
+
+            var templateModel = new BlazorModel(modelTypeAndContextModel.ModelType, modelTypeAndContextModel.DbContextFullName)
+            {
+                Namespace = modelTypeAndContextModel.ModelType.Namespace,
+                ModelMetadata = modelTypeAndContextModel.ContextProcessingResult?.ModelMetadata,
+                DatabaseProvider = model.DatabaseProvider,
+                Template = model.TemplateName
+            };
+
+            ExecuteTemplates(templateModel);
+            await ModifyProgramCsAsync();
+        }
+
+        internal IList<string> GetT4Templates(string templateName)
+        {
+            var templates = new List<string>();
+            var crudTemplate = templateName.Equals("crud", StringComparison.OrdinalIgnoreCase);
```

Comment:
I'd use `string.Equals()`. If `templateName` is `NULL`, this code will throw

Reply:
sounds good ty